specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json

```diff
@@ -2854,6 +2905,52 @@
           "type": "object"
         }
       },
+      "FactoryVSTSConfiguration": {
+        "description": "Factory's VSTS repo information.",
+        "properties": {
+          "accountName": {
+            "description": "VSTS account name.",
+            "type": "string"
+          },
+          "projectName": {
+            "description": "VSTS project name.",
+            "type": "string"
+          },
+          "repositoryName": {
+            "description": "VSTS repository name.",
+            "type": "string"
+          },
+          "collaborationBranch": {
+            "description": "VSTS collaboration branch.",
+            "type": "string"
+          },
+          "rootFolder": {
+            "description": "VSTS root folder.",
+            "type": "string"
+          },
+          "lastCommitId": {
+            "description": "VSTS last commit id.",
+            "type": "string"
+          },
+          "tenantId": {
+            "description": "VSTS tenant id.",
+            "type": "string"
+          }
+        }
+      },
+      "FactoryRepoUpdate": {
+        "description": "Factory's VSTS repo information.",
+        "properties": {
+          "factoryId": {
+            "description": "The factory id.",

```

Comment:
This is the full ARM resource ID for the factory right? Since your API is under factory resource now, you shouldn't need this, since all information will be in your path.

Reply:
I need this property for the linkedActionCheck

Reply:
You can construct it since you have the factory name, rg name, sub id in your headers when the API gets called. We shouldn't require the client to send this extra field if we already have all the information.